gradle-plugins/model-compiler/src/main/java/io/spine/gradle/compiler/lookup/valrule/ValidationRulesFinder.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.gradle.compiler.lookup.valrule;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.DescriptorProtos.DescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import org.slf4j.Logger;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Maps.newLinkedHashMap;
+import static io.spine.gradle.compiler.util.UnknownOptions.getUnknownOptionValue;
+import static io.spine.gradle.compiler.util.UnknownOptions.hasUnknownOption;
+import static io.spine.option.OptionsProto.VALIDATION_OF_FIELD_NUMBER;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Finds Protobuf definitions of validation rules.
+ *
+ * @author Dmytro Grankin
+ */
+class ValidationRulesFinder {
+
+    @VisibleForTesting
+    static final String PROTO_TYPE_SEPARATOR = ".";
+
+    private final FileDescriptorProto fileDescriptor;
+    private final String packagePrefix;
+
+    ValidationRulesFinder(FileDescriptorProto fileDescriptor) {
+        this.fileDescriptor = checkNotNull(fileDescriptor);
+        this.packagePrefix = fileDescriptor.getPackage() + PROTO_TYPE_SEPARATOR;
+    }
+
+    /**
+     * Finds Protobuf definitions of validation rules in the file descriptor.
+     *
+     * @return a map from a validation rule type name to the target field
+     */
+    Map<String, String> findRules() {
```

Comment:
Why cannot this be a static method with a parameter? At the moment the fact that this class has a state does not look OK.

I would even consider turning this class (with a bit of renaming) a `static` nested class of `ValidationRules`.